### PR TITLE
Feat: integration test environment redeploy

### PR DESCRIPTION
**Pull request by meniRoy**
Integration tests for environment redeploy.

- Ensure that HTTP requests for redeploy are called with the correct arguments and authentication.
- Verify that pop-up notifications display the appropriate message when the environment status is updated.
- Confirm that the "show more" button in the pop-up notification opens the environment page in the browser.
- Validate that the environment, its icon, and status are updated in line with the environment's status.

I haven't included tests for other environment actions  in this PR. A separate PR will be opened for that.

For stubs and mocks, I've utilized jest-mock.

Given the limitations where I can't simulate a user click on an environment action, I've opted to execute the command that's triggered when a user clicks on an environment action. 
Additionally, since I can't retrieve the value from a VS Code pop-up notification directly, I've employed a spy on the VS Code API responsible for displaying the pop-up notification.

### CODE REVIEW

**Comment by meniRoy** [comment]
[Integration Tests](https://trello.com/c/eoV0MdoJ/10053-integration-tests)

File: src/test/integration/suite/test-utils.ts [hunk 1]
```diff
@@ -36,3 +46,78 @@ export const getEnvironmentMock = (
...
+}
+
+export const stubShowMessage = (messageType: MessageType) => {
+  const openExternalMock = jestMock.spyOn(vscode.env, "openExternal");
+  const mockUriParse = jestMock
+    .spyOn(vscode.Uri, "parse")
+    .mockImplementation((url) => url as any);
+  openExternalMock.mockResolvedValue(true);
+  const showMessageMock = jestMock.spyOn(vscode.window, messageType);
+  showMessageMock.mockResolvedValue("More info" as any);
```
Comment (liranfarage89) [inline src/test/integration/suite/test-utils.ts:63]: @meniRoy I'm not sure I'm following what's going on here.
you mock `showMessageMock.mockResolvedValue("More info" as any);`

then you verify that it was called with "more info"? 🤔

File: src/test/integration/suite/environment-actions.test.it.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    await logout();
+    await resetExtension();
+  });
+
+  suite("redeploy", () => {
+    test("should redeploy when user redeploy", async () => {
+      await initTest([environmentMock]);
+
+      const onRedeployCalled = jestMock.fn();
+      mockRedeploy(environmentMock.id, auth, onRedeployCalled);
```
Comment (liranfarage89) [inline src/test/integration/suite/environment-actions.test.it.ts:85]: ``
```suggestion
      mockRedeployApiResponse(environmentMock.id, auth, onRedeployCalled);
```

File: src/test/integration/suite/environment-actions.test.it.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        orgId,
+      });
+
+      expect(getFirstEnvIconPath()).toContain(inProgressIconPath);
+      expect(getFirstEnvStatus()).toBe("DEPLOY_IN_PROGRESS");
+
+      await mockEnvironmentWithUpdatedStatus(
+        inProgressEnvironment,
+        EnvironmentStatus.ACTIVE
+      );
```
Comment (liranfarage89) [inline src/test/integration/suite/environment-actions.test.it.ts:105]: This looks great 👍

File: src/test/integration/suite/environment-actions.test.it.ts (hunk 2 above)
Comment (meniRoy) [reply src/test/integration/suite/environment-actions.test.it.ts]: 59e816e43322fafef360f289a5bf7722d6e8f973

File: src/test/integration/suite/test-utils.ts (hunk 1 above)
Comment (meniRoy) [reply src/test/integration/suite/test-utils.ts]: 9bc46ec54402604d0b203a9d270e72d977906c35

File: src/test/integration/suite/environment-actions.test.it.ts (hunk 3 above)
Comment (meniRoy) [reply src/test/integration/suite/environment-actions.test.it.ts:135]: 👍

**Review by liranfarage89 — APPROVED** [review]
Well done @meniRoy 👏 
please remove sinon leftovers before merging

**Pull request merged**